src/test/java/net/mcreator/integration/TestWorkspaceDataProvider.java

```diff
@@ -1127,6 +1119,19 @@ null, null, new MItemBlock(modElement.getWorkspace(),
 					getRandomMCItem(random, ElementUtil.loadBlocksAndItems(modElement.getWorkspace())).getName());
 			item.onFinishUsingItem = new Procedure("procedure3");
 			return item;
+		} else if (ModElementType.ITEMEXTENSION.equals(modElement.getType())){
+			ItemExtension itemExtension = new ItemExtension(modElement);
+			itemExtension.item = new MItemBlock(modElement.getWorkspace(), getRandomMCItem(random, blocksAndItems).getName());
+
+			itemExtension.enableFuel = !emptyLists;
+			itemExtension.fuelPower = new NumberProcedure(emptyLists ? null : "number1", 1600);

```

Comment:
This is not ok. !emptyLists condition fuel generation, then emptyLists conditions procedures, so procedures are never tested